_posts/guides/2021-12-21-guide_gamemode-info-bhop.md

```diff
@@ -19,14 +19,32 @@ toc_sticky: true
+
 ## History of Bhop
 
-TODO full history of bhop, mentioning defrag probably as the grandpa
+**TODO:** full history of bhop, mentioning defrag probably as the grandpa. Some paragraphs might have to be relocated here, I'm not totally sure where each explanation should be.
+
+## How Bhop Works (Non-Technical)
+
+Normally when the player hits the ground, they slow down from friction. If the player instead jumps the exact frame they hit the ground, they will maintain their speed. Players can then combine this with air-strafing to accelerate faster and faster with each consecutive hop.
+
+Air-strafing is achieved by hitting the keyboard directional button corresponding to the direction the player is moving their mouse. The player will then gain momentum in the direction they're facing, even as the direction they're facing changes. In order to move quickly in a straight line, players keep alternating directions. With that in mind the sequence of events looks something like this:
+
+1. Jump off the ground
+2. Air-strafe left
+3. Perform a Bhop (auto or traditional)
+4. Air-strafe right
+
+This allows players to gain speed from air strafing while moving in a straight line.
+
+There is a nearly endless amount of complexity in doing this to the greatest effect. The optimal speed for your mouse changes with forward velocity, there are times where it may be optimal to switch directions multiple times in one jump, speed can be gained by landing on a downward facing ramp (especially if falling from a height), and many other techniques can go into getting the best times on a specific map.
+
+## Technical Explanation
+
+Movement in source is made up of a player's orientation, velocity, and acceleration (each represented by a vector). The acceleration vector is determined by the orientation and which directional keys the player is pressing (the rate of air acceleration is generally about 10 times the player's normal maximum walking speed per second, which is what makes characters feel responsive in mid-air).
 
-## How Bhop Works
+Now, in order to prevent players from gaining infinite speed simply by holding W and space, the source engine adds an air-speed limit (not to be confused with sv_airaccelerate, sv_maxspeed, or sv_maxacceleration). This air speed limit isn't an absolute cap, rather each tick the player's velocity is projected onto their acceleration. If the magnitude of this projection is less than the speed limit, the acceleration is applied clamped to the limit. If this projection is greater than the speed limit, acceleration is ignored. This means that for acceleration vectors very close to the velocity vector, acceleration is almost always ignored if the magnitude of the velocity vector is greater than the speed cap. To put it in perspective, the default air speed limit 30 HU/s. In CS:GO, a player holding the Negev (the heaviest weapon) runs at 150 HU/s, and in tf2 a heavy with a revved-up minigun runs at 110 HU/s.
```

Comment:
I'd swap the comparison from any minigun to the Brass Beast which is 60% slower (44HU/s) then the stock minigun https://wiki.teamfortress.com/wiki/Heavy#Speed